### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by laurachiticariu**
Clarify system requirements upfront.

### CODE REVIEW

File: README.md
Comment (joesepi) [inline README.md:80]:
```suggestion
- **🍎 Apple M1/M2/M3 Mac or 🐧 Linux system** (tested on Fedora). Note Linux
 is only partially supported (serving and testing a trained model does not work on Linux).
  We anticipate support for more operating systems in the future.
```

**Review by joesepi — APPROVED** [review]
LGTM. Thanks!

File: README.md
Comment (chudegao) [inline README.md:81]: I think serving is supported already,  test and convert is not supported on linux so far.  There's a [PR](https://github.com/instructlab/instructlab/pull/1337) under review for test support. Convert is not needed for linux after train as trained model is a converted model already.

Comment (leseb) [reply README.md:81]: Indeed serving is supported.

**Review by leseb — CHANGES_REQUESTED** [review]
Please rephrase based on comment reviews. Thanks

File: README.md
Comment (laurachiticariu) [reply README.md:81]: I have learned from Akash Srivastava that while test and serve is supported for the base models, it is not supported for models trained through `ilab train` today. You are saying there is a PR in the works to support this. Once the [PR you mention](https://github.com/instructlab/instructlab/pull/1337) is merged, it should change this readme accordingly. Until then, I'd appreciate if you merge this to avoid confusing users who are testing this now.

Comment (bjhargrave) [reply README.md:81]: After reading through all the discussion here, I think the proposed text is valid and correct:

> Note Linux is only partially supported (serving and testing a trained model does not work on Linux).

So I think we can merge this for now and update later (along with https://github.com/instructlab/instructlab/blob/main/README.md#-test-the-newly-trained-model) when support is added to allow linux to serve and test an ilab trained model.

Comment (chudegao) [reply README.md:81]: serve a trained model on linux is supported. Here's an example:
```
(dlipy3) [root@llm chu]# ilab serve --model-path models/ggml-model-f16.gguf
You are using an aliased command, this will be deprecated in a future release. Please consider using `ilab model serve` instead
INFO 2024-06-25 04:57:20,095 serve.py:56: serve Using model 'models/ggml-model-f16.gguf' with -1 gpu-layers and 4096 max context size.
INFO 2024-06-25 04:57:22,738 llama.py:268: server Starting server process, press CTRL+C to shutdown server...
INFO 2024-06-25 04:57:22,738 llama.py:269: server After application startup complete see http://10.244.128.4:4001/docs for API.
```
models/ggml-model-f16.gguf is trained model on linux.

ilab model convert is only need for mac. Maybe some updates are needed in 'Serve the newly trained model' part to make mac and linux serve more clear.

Comment (laurachiticariu) [reply README.md:81]: Ok, so training a model on Linux is not supported on Linux. But serving a trained model is supported on Linux? Did I get this right? So the full life cycle is still not supported on Linux. Since I don't have a Linux machine to verify this statement, how do you all propose to modify this PR to ensure the statements are accurate? Can someone with knowledge of the code actually refresh this PR with the accurate wording so users don't get confused? Since 5 days ago when I proposed this change, I got another 10 users who complained the top level statement on Linux support is misleading.

**Review by markstur — APPROVED** [review]
This is at least an improvement.

File: README.md
Comment (chudegao) [reply README.md:81]: training on linux is supported also.

Comment (chudegao) [reply README.md:81]: But serving a trained model is supported on Linux?   -- Yes. serveing is supported.  

So the full life cycle is still not supported on Linux.   --- I don't think so. Actually, testing is part of training on linux, you can see the test result when do training.(just can not do test separately).

Comment (chudegao) [reply README.md:81]: But serving a trained model is supported on Linux?   -- Yes. serving is supported.  

So the full life cycle is still not supported on Linux.   --- I don't think so. Actually, testing is part of training on linux, you can see the test result when do training.(just can not do test separately).

Comment (joesepi) [inline README.md:81]: Is this accurate? If so, can we get the block lifted @leseb @chudegao 🙏 

```suggestion
- **🍎 Apple M1/M2/M3 Mac or 🐧 Linux system** (tested on Fedora). Note Linux
 is not fully supported (testing a trained model does not currently work on Linux).
```

Comment (bjhargrave) [reply README.md:81]: This will need to be added to the PR before approval.

**Comment by joesepi** [comment]
@leseb -- I've committed my suggested change in the hopes that it accurately reflects the current state of the project and can be merged. Please and thank you! :)

**Review by joesepi — APPROVED** [review]

**Comment by leseb** [comment]
@joesepi for next time, it'd be nice to squash commits, thanks!

**Comment by nathan-weinberg** [comment]
I can squash real quick

**Comment by laurachiticariu** [comment]
Thank you all!

**Pull request merged**
